src/activity_item_handling.cpp

```diff
@@ -1200,64 +1244,113 @@ static activity_reason_info can_do_activity_there( const activity_id &act, playe
+                for( vehicle_part *part_elem : parts ) {
+                    const vpart_info &vpinfo = part_elem->info();
+                    int vpindex = veh->index_of_part( part_elem, true );
+                    // if part is not on this vehicle, or if its attached to another part that needs to be removed first.
+                    if( vpindex == -1 || !veh->can_unmount( vpindex ) ) {
                         continue;
                     }
-                }
-                // if the vehicle is moving or player is controlling it.
-                if( abs( veh->velocity ) > 100 || veh->engine_on ) {
-                    continue;
-                }
-                // dont have skill to remove it
-                std::map<skill_id, int> removal_skills = vpinfo.removal_skills;
-                for( const auto &e : removal_skills ) {
-                    bool hasSkill = p.get_skill_level( e.first ) >= e.second;
-                    if( !hasSkill ) {
+                    for( const npc &guy : g->all_npcs() ) {
+                        if( guy.disp_name() != p.disp_name() && guy.activity_vehicle_part_index != -1 &&
+                            guy.activity_vehicle_part_index == vpindex ) {
+                            continue;
+                        }
+                    }
+                    // if the vehicle is moving or player is controlling it.
+                    if( abs( veh->velocity ) > 100 || veh->engine_on ) {
                         continue;
                     }
+                    // dont have skill to remove it
+                    std::map<skill_id, int> removal_skills = vpinfo.removal_skills;
+                    for( const auto &e : removal_skills ) {
+                        bool hasSkill = p.get_skill_level( e.first ) >= e.second;
+                        if( !hasSkill ) {
+                            continue;
+                        }
+                    }
+                    item base( vpinfo.item );
+                    if( base.is_wheel() ) {
+                        // no wheel removal yet
+                        continue;
+                    }
+                    const quality_id qual = LIFT;
+                    const int max_lift = p.best_nearby_lifting_assist( src_loc );
+                    const int lvl = ceil( units::quantity<double, units::mass::unit_type>( base.weight() ) /
+                                          TOOL_LIFT_FACTOR );
+                    const bool use_aid = max_lift >= lvl;
+                    const bool use_str = p.can_lift( base );
+                    if( !( use_aid || use_str ) ) {
+                        continue;
+                    }
+                    const auto &reqs = vpinfo.removal_requirements();
+                    const std::string ran_str = random_string( 10 );
+                    const requirement_id req_id( ran_str );
+                    requirement_data::save_requirement( reqs, req_id );
```

Comment:
What does that even attempt to do? The requirement is stored in the list of all requirements using a random id.

But that id is never stored anywhere else. So there is no way of getting this requirement back (the ID is random!).